Changes and fixes to MyEID support

hhonkanen opened the pull request:
Fixed bugs related to handling ECC key size when creating/importing keys and added support for 521 bit keys.

frankmorgner commented on .gitignore:
```diff
@@ -1,122 +0,0 @@
-Makefile
```
did you accidentally delete this file?

hhonkanen replied:
Yes, I had some confusion with this file, because I have just started using git recently. I thought it was something accidentally added from my own file system and it prevented me from merging the master into my fork, so I ended up removing it. My intention was to not change this file in any way, so please ignore this change.

frankmorgner replied:
Please restore `.gitignore` and rebase onto the current HEAD

hhonkanen replied:
Did this, please could you check is it ok now?

frankmorgner commented on src/pkcs15init/pkcs15-myeid.c:
```diff
@@ -431,7 +431,16 @@ myeid_create_key(struct sc_profile *profile, struct sc_pkcs15_card *p15card,
-			   without curve OID. */
+			   without curve OID. 
+			  
+			   Added setting keybits from key_info->field_length -Hannu Honkanen
```
please use a clear comment that does not require your name

frankmorgner commented on src/pkcs15init/pkcs15-myeid.c:
```diff
@@ -497,13 +506,16 @@ myeid_store_key(struct sc_profile *profile, struct sc_pkcs15_card *p15card,
-			if (sc_card_find_ec_alg(p15card->card, keybits, &prkey->u.ec.params.id) == NULL)
-				LOG_TEST_RET(ctx, SC_ERROR_INVALID_ARGUMENTS, "Unsupported algorithm or key size");
+			
+			 /* moved setting keybits BEFORE calling sc_card_find_ec_alg - Hannu Honkanen */
```
please use a clear comment that does not require your name

frankmorgner commented on src/pkcs15init/pkcs15-myeid.c:
```diff
@@ -587,12 +599,19 @@ myeid_generate_key(struct sc_profile *profile, struct sc_pkcs15_card *p15card,
-				LOG_TEST_RET(ctx, SC_ERROR_INVALID_ARGUMENTS, "Unsupported EC key size");
+			
+			/*
+                         * set keybits BEFORE calling sc_card_find_ec_alg - Hannu Honkanen 30.10.2015
```
please use a clear comment that does not require your name

frankmorgner commented on src/libopensc/card-myeid.c:
```diff
@@ -802,6 +845,9 @@ myeid_compute_signature(struct sc_card *card, const u8 * data, size_t datalen,
 	struct myeid_private_data* priv;
 	int r;
+	unsigned int field_length = 0; 
+	int pad_chars = 0;
```
please use `size_t` for `field_length` and `pad_chars `

frankmorgner commented on src/libopensc/card-myeid.c:
```diff
@@ -768,11 +784,38 @@ myeid_convert_ec_signature(struct sc_context *ctx, size_t s_len, unsigned char *
-
-	if (*data != 0x30 || *(data + 1) != (datalen - 2) || *(data + 2) != 0x02)
+	unsigned int len_size = 1;
+	unsigned int sig_len = 0;
```
please use `size_t` for `len_size` and `sig_len `

frankmorgner commented:
despice the formal remarks (see above) the change looks OK.

Please add a commit that removes the define/check for `MYEID_ECC_SUPPORT` entirely, because it is not used in any usable way.

hhonkanen commented:
Frank, thank you for reviewing this pull request. I have updated the code according to your notes and pushed the changes to my fork. I added also a new ATR. Should I create a new pull request now or will my changes be merged to this pull request? 
BTW, I tried to post to the mailing list about my commits, but posting failed because our e-mail operator has changed since I posted there last time and I have subscribed the list under a different account that I normally use. I will get this fixed soon.

frankmorgner commented on src/libopensc/card-myeid.c:
```diff
@@ -768,11 +776,38 @@ myeid_convert_ec_signature(struct sc_context *ctx, size_t s_len, unsigned char *
+	/*
+		more advanced validation is needed, because with 521 bit keys
+		length of the signature TLV is encoded in two bytes.
+		- Hannu Honkanen 6.11.2015
```
Should this be a "`TODO`"?

Please remove your name and the date here (it will be present in the git history)

hhonkanen replied:
It's not a TODO, it has been implemented already. Cleaned up and rewrote this comment.

frankmorgner commented:
thanks! Merged with some commit cleanup
